Apps/Admin/Client/Components/Delegation/DelegationConfirmationDialog.razor.cs

```diff
@@ -95,6 +101,8 @@ private void HandleClickCancel()
 
     private void HandleClickConfirm()
     {
+        this.DelegationState.Value.Reason = this.ProtectReasonString;
```

Comment:
The state shouldn't be modified except by mutations that trigger when actions are dispatched.

An easy fix here is to update ProtectDependentAction and UnprotectDependentAction to have a property for the reason.  It can be populated here and accessed directly in the effects.